### PR TITLE
Fix: SCIM error handling for group display name (fixes #20344)

### PR DESCRIPTION
**Pull request by BimsaraBodaragama**
## Purpose
Fix SCIM error handling to provide appropriate messages when a group display name is used instead of a group ID.

## Goals
To ensure compliance with RFC 7644 and provide clear, actionable error messages to users when an invalid request is made with a group display name.

## Approach
- Implemented error response with `scimType` set to `invalidSyntax` for requests using a group display name instead of a group ID.
- Provided a clear and concise error message in the `detail` field.
- Ensured the error response adheres to the SCIM specification.

## Issue
Resolves [#20334](https://github.com/wso2/product-is/issues/20334)

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java [hunk 1]
```diff
@@ -1125,7 +1125,13 @@ private void doUpdateGroups(String roleId, Set<String> newGroupIDList, Set<Strin
                         new ArrayList<>(deleteGroupIDList), tenantDomain);
             } catch (IdentityRoleManagementException e) {
                 if (RoleConstants.Error.INVALID_REQUEST.getCode().equals(e.getErrorCode())) {
-                    throw new BadRequestException(e.getMessage());
+                    // Error message and SCIM type
+                    String errorMessage = "Updating groups of the role by display name is not supported. "
+                            + "Update using group id instead.";
+                    String scimType = "invalidSyntax"; // From RFC 7644 Table 9
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java:1131]: Without hardcoding here you can directly use ResponseCodeConstants.INVALID_SYNTAX constant in charon

scimType variable is also redundant then

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java [hunk 2]
```diff
@@ -1125,7 +1125,13 @@ private void doUpdateGroups(String roleId, Set<String> newGroupIDList, Set<Strin
                         new ArrayList<>(deleteGroupIDList), tenantDomain);
             } catch (IdentityRoleManagementException e) {
                 if (RoleConstants.Error.INVALID_REQUEST.getCode().equals(e.getErrorCode())) {
-                    throw new BadRequestException(e.getMessage());
+                    // Error message and SCIM type
+                    String errorMessage = "Updating groups of the role by display name is not supported. "
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java:1129]: Is this the correct place to handle the mentioned error?
Issue should be directly taking `display eq ...` something as group id 
We have to throw the error from that place

Comment (BimsaraBodaragama) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java]: Addressed in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/557/commits/b974ba9606102966ce3ceb702105778fa9623348

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManager.java [hunk 3]
```diff
@@ -771,21 +771,30 @@ private void updatePermissions(String roleId, List<PatchOperation> permissionOpe
 
     private void prepareAddedRemovedGroupLists(Set<String> addedGroupsIds, Set<String> removedGroupsIds,
                                                Set<String> replacedGroupsIds, PatchOperation groupOperation,
-                                               Map<String, String> groupObject, List<GroupBasicInfo> groupListOfRole) {
+                                               Map<String, String> groupObject, List<GroupBasicInfo> groupListOfRole)
+            throws BadRequestException {
+
+        String value = groupObject.get(SCIMConstants.CommonSchemaConstants.VALUE);
+
+        if (value == null) {
```
Comment (ShanChathusanda93) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManager.java:779]: Shall we use the StringUtils?

Comment (BimsaraBodaragama) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManager.java]: Sure

**Pull request closed without merging**